### PR TITLE
fix issue, when labels are not added to new node pools in case there …

### PR DESCRIPTION
**Pull request by sancyx**
…are no custom labels specified

| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #X, partially #Y, mentioned in #Z
| License         | Apache 2.0

### What's in this PR?
- remove POST nodepool-labels API call as it's not used
- fix issue, no labels are present on a newly added node in case there were no custom labels specified in the request

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)

### CODE REVIEW

File: cmd/pipeline/main.go [hunk 1]
```diff
@@ -827,7 +827,6 @@ func main() {
 			cgroupsAPI.AddRoutes(orgs.Group("/:orgid/clustergroups"))
 
 			cRouter.GET("/nodepool-labels", nplsApi.GetNodepoolLabelSets)
-			cRouter.POST("/nodepool-labels", nplsApi.SetNodepoolLabelSets)
```
Comment (stoader) [inline cmd/pipeline/main.go]: Is this not used any more?

File: internal/providers/amazon/eks/driver/cluster_creator.go [hunk 2]
```diff
@@ -190,11 +192,24 @@ func (c *EksClusterCreator) create(ctx context.Context, logger logrus.FieldLogge
 			Labels:           np.Labels,
 		}
 		asgList = append(asgList, asg)
+
+		nodePoolLabels = append(nodePoolLabels, cluster.NodePoolLabels{
```
Comment (stoader) [inline internal/providers/amazon/eks/driver/cluster_creator.go]: The `NodePoolLabels ` name is misleading as it seems that beside Labels it holds node pool related fields as well.

File: src/api/npls.go [hunk 3]
```diff
@@ -115,102 +112,3 @@ func (n *NodepoolManagerAPI) GetNodepoolLabelSets(c *gin.Context) {
 
 	c.JSON(http.StatusOK, response)
 }
-
-func (n *NodepoolManagerAPI) SetNodepoolLabelSets(c *gin.Context) {
```
Comment (stoader) [inline src/api/npls.go]: Isn't this used any more?

Comment (sancyx) [reply src/api/npls.go]: it not used by UI as labels are updated in cluster update request

File: cmd/pipeline/main.go (hunk 1 above)
Comment (sancyx) [reply cmd/pipeline/main.go]: it not used by UI as labels are updated in cluster update request

File: internal/providers/amazon/eks/driver/cluster_creator.go (hunk 2 above)
Comment (sancyx) [reply internal/providers/amazon/eks/driver/cluster_creator.go]: Actually it holds some fields from node pool like name, instance type & spot price, necessary for cloud info call.

File: src/cluster/nodelabels.go [hunk 4]
```diff
@@ -26,17 +26,25 @@ import (
...
 
 const labelFormatRegexp = "[^-A-Za-z0-9_.]"
 
+type NodePoolLabels struct {
+	Name         string
+	Existing     bool
+	InstanceType string            `json:"instanceType,omitempty"`
+	SpotPrice    string            `json:"spotPrice,omitempty"`
+	Preemptible  bool              `json:"preemptible,omitempty"`
+	Labels       map[string]string `json:"labels,omitempty"`
```
Comment (stoader) [inline src/cluster/nodelabels.go]: I'd rename this to something like `CustomLabels` or `ExtraLabels` to make it clear that these are user provided labels while the others are set by Pipeline.

File: src/cluster/nodelabels.go [hunk 5]
```diff
@@ -26,17 +26,25 @@ import (
...
 	"github.com/banzaicloud/pipeline/internal/global"
 	pipelineContext "github.com/banzaicloud/pipeline/internal/platform/context"
-	pkgCluster "github.com/banzaicloud/pipeline/pkg/cluster"
 	"github.com/banzaicloud/pipeline/pkg/common"
 )
 
 const labelFormatRegexp = "[^-A-Za-z0-9_.]"
 
+type NodePoolLabels struct {
+	Name         string
```
Comment (stoader) [inline src/cluster/nodelabels.go]: `Name` -> `NodePoolName`

Comment (sancyx) [reply src/cluster/nodelabels.go]: renamed

File: src/cluster/nodelabels.go (hunk 4 above)
Comment (sancyx) [reply src/cluster/nodelabels.go]: renamed to CustomLabels

**Pull request merged**